### PR TITLE
doc: relnotes: Add Bluetooth release notes for 3.2.0

### PR DESCRIPTION
**Pull request by carlescufi**
Populate the Bluetooth release notes and clean up the existing ones.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

**Review by mtpr-ot — DISMISSED** [review]
Looks fine to me, from controller perspective

File: doc/releases/release-notes-3.2.rst [hunk 1]
```diff
@@ -223,27 +223,95 @@ Bluetooth
...
+  * :c:func:`bt_conn_get_security` and :c:func:`bt_conn_enc_key_size` now take
+    a ``const struct bt_conn*`` argument
+  * The handling of GATT multiple notifications has been rewritten, and is now
+    only to be used as a low-level API
+  * Added support for GATT CCCs in arbitrary locations as a client
+  * Extended the ``bt_conn_info`` structure with security information
+  * Added a new :kconfig:option:`CONFIG_BT_PRIVACY_RANDOMIZE_IR` that prevents
+    the Host from using Controller-provided identity roots
+  * Added support for GATT over EATT
+  * Implemented the Immediate Alert Client
```
Comment (jori-nordic) [inline doc/releases/release-notes-3.2.rst:278]: I think there's also the channel selection option for EATT. Don't know if we should mention it here, what do you think @hermabe ?

Comment (hermabe) [reply doc/releases/release-notes-3.2.rst:278]: >  * Added support for GATT over EATT

Is it not mentioned on the line above?

However, I think I messed up when merging the note in the release note and it was added to the 3.1 release notes. The PR was opened before 3.1 was released, but merged afterwards.
https://github.com/zephyrproject-rtos/zephyr/blob/2cea5850e2cd7587858276da9d8c84ee90568824/doc/releases/release-notes-3.1.rst?plain=1#L28-L30

We should move it to the 3.2 release notes.

**Review by kapi-no — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

File: doc/releases/release-notes-3.2.rst [hunk 2]
```diff
@@ -223,27 +223,95 @@ Bluetooth
...
 
 * Mesh
 
+  * Added support for selectable RPL backends
+  * Changed the way segmented messages are sent, avoiding bulk transmission
+  * Added an async config client API
+  * Added model publish support to the Health Client
+  * Moved relayed messages to a separate buffer pool
```
Comment (PavelVPV) [inline doc/releases/release-notes-3.2.rst:286]: I would suggest to add:
```
* Reduced delay of sending segment acknowledge message. Set :kconfig:option:`CONFIG_BT_MESH_SEG_ACK_PER_SEGMENT_TIMEOUT` to 100 to get previous timing.
* Restructured shell commands
```

Comment (carlescufi) [reply doc/releases/release-notes-3.2.rst:286]: Done

File: doc/releases/release-notes-3.2.rst (hunk 1 above)
Comment (carlescufi) [reply doc/releases/release-notes-3.2.rst:282]: I added it to the 3.2 release notes. Left it in the 3.1 ones to avoid confusion.

**Review by PavelVPV — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**